tasks/Module 4/task_2.py

```diff
@@ -0,0 +1,39 @@
+"""Task 2 of Module 4."""
+from pathlib import Path
+import os
+from tkinter.filedialog import askdirectory
+
+
+def filling_in_the_gaps():
```

Comment:
It does not work as expected for some cases. Please test it with following input files: `spam001.txt`, `spam002.txt`, `spam004.txt`, `spam007.txt`. Expected result is `spam001.txt`, `spam002.txt`, `spam003.txt`, `spam004.txt`, but instead I got `spam002.txt`, `spam003.txt`, `spam004.txt`.

Reply:
@musk-ox strange bu it works for me. I checked several times and the fiels from `spam001.txt, spam002.txt, spam004.txt, spam007.txt` got renamed to `spam001.txt, spam002.txt, spam003.txt, spam004.txt`

Reply:
Weird, with old version of code I was getting wrong results. Maybe it was because of lack of sorting and `os.listdir` return results in different order depending on OS. Anyway, it works now 👍